src/components/Map/index.js

```diff
@@ -82,6 +102,32 @@ const MapView = ({coordinates = null, zoomLevel = DEFAULT_ZOOM}) => {
   return null;
 };
 
+const SearchControl = props => {
+  const map = useMap();
+  const onMarkerSet = props.onMarkerSet;
+
+  const setMarker = result => {
+    if (result.location) {
+      onMarkerSet([result.location.y, result.location.x]);
+    }
+  };
+
+  useEffect(() => {
+    const searchControl = new GeoSearchControl({
+      provider: props.provider,
+      style: 'button',
+      ...props,
+    });
```

Comment:
Can we investigate if we can modify this to add aria-label attributes where required?

Looking at the [source code](https://github.com/smeijer/leaflet-geosearch/blob/develop/src/SearchControl.ts#L487), it should be possible to do something like:

```js
searchControl.button.setAttribute("aria-label", intl.formatMessage(...)); // set accessible label
```
we need to make sure that `initialize` has run though, that might only be the case after the `map.addControl` call?

Reply:
And I think a components `props`  are always implicitly a dependency, so don't have to be added to the dependency list of hooks.

Reply:
no, that's not right - but it would be better to destructure the props and make explicit what is being passed along.

While the individual props may stay the same, this is not guaranteed for the object exposing them, and Javascript/hook dependency comparison does not work with deep equality but with object identity. So this would trigger the hook every time the component re-renders _if React doesn't memoize the props itself_.